core/field_input.ts

```diff
@@ -499,7 +521,20 @@ export abstract class FieldInput<T extends InputTypes> extends Field<
    * @param _e Keyboard event.
    */
   private onHtmlInputChange_(_e: Event) {
+    // Intermediate value changes from user input are not confirmed until the
+    // user closes the editor, and may be numerous. Inhibit reporting these as
+    // normal block change events, and instead report them as special
+    // intermediate changes that do not get recorded in undo history.
+    const oldValue = this.value_;
+    eventUtils.disable();
     this.setValue(this.getValueFromEditorText_(this.htmlInput_!.value));
+    eventUtils.enable();
```

Comment:
I don't love the use of `disable` and `enable` to change the behavior here, because we're using the state of a separate system (the event system) to change the behavior of the super class (the `Field`). I think this makes the logic harder to follow/debug.

Do you think it would be possible / how would you feel about passing a flag to `setValue` instead? Similarly to your previous implementation: https://github.com/google/blockly/pull/7064/files#diff-a671fdac0ad6e251056d1b8aa6f3e2d746d3476e8945ff81624f5733dc4e28e4R996

Reply:
Done!